src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/visitbalances/VisitBalanceSynchronisationService.kt

```diff
@@ -1,68 +1,149 @@
 package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.visitbalances
 
 import com.microsoft.applicationinsights.TelemetryClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.config.trackEvent
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.helpers.telemetryOf
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.helpers.trackEvent
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.helpers.valuesAsStrings
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.listeners.SynchronisationMessageType
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomismappings.model.VisitBalanceAdjustmentMappingDto
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomisprisoner.model.VisitBalanceAdjustmentResponse
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomisprisoner.model.VisitBalanceDetailResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.InternalMessage
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.NomisApiService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.SynchronisationQueueService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.SynchronisationType
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.visit.balance.model.VisitAllocationPrisonerSyncDto
 
 @Service
 class VisitBalanceSynchronisationService(
   private val nomisVisitBalanceApiService: VisitBalanceNomisApiService,
   private val nomisApiService: NomisApiService,
   private val dpsApiService: VisitBalanceDpsApiService,
+  private val mappingApiService: VisitBalanceMappingApiService,
+  private val queueService: SynchronisationQueueService,
   private val telemetryClient: TelemetryClient,
 ) {
   private companion object {
     const val VISIT_ALLOCATION_SERVICE = "VISIT_ALLOCATION"
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
   suspend fun visitBalanceAdjustmentInserted(event: VisitBalanceOffenderEvent) {
     val visitBalanceAdjustmentId = event.visitBalanceAdjustmentId
     val nomisPrisonNumber = event.offenderIdDisplay
-    val telemetry = telemetryOf("visitBalanceAdjustmentId" to visitBalanceAdjustmentId, "nomisPrisonNumber" to nomisPrisonNumber)
+    val telemetry =
+      telemetryOf("nomisVisitBalanceAdjustmentId" to visitBalanceAdjustmentId, "dpsId" to nomisPrisonNumber)
```

Comment:
this looks weird - we're creating telemetry for a prison number and calling it `dpsId`?

Reply:
will swap back